chroma_feedback/consumer/razer_chroma/device.py

```diff
@@ -68,5 +76,14 @@ def pulsate_device(device : Any, state : Dict[str, Any]) -> bool:
 	if device.has('brightness'):
 		device.brightness = state['brightness'][0]
 	if device.fx.has('logo') and device.fx.has('scroll'):
-		return device.fx.misc.logo.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2]) and device.fx.misc.scroll_wheel.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2])
+		if device.fx.has('logo_pulsate') and device.fx.has('scroll_pulsate'):
+			return device.fx.misc.logo.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2]) and device.fx.misc.scroll_wheel.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2])
+	return device.fx.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2])
+
+def breath_device(device : Any, state : Dict[str, Any]) -> bool:
+	if device.has('brightness'):
+		device.brightness = state['brightness'][0]
+	if device.fx.has('logo') and device.fx.has('scroll'):
```

Comment:
I think we don't need that double checks...`logo_breath_single` and `scroll_breath_single` should only be positive when `logo` and `scroll` even exist. Same for `pulsate_device()` method...

Please check if there `logo_static` and `scroll_static` has check for the `static_device()` method.

Reply:
Sure, I can make that change.  (I was mostly concerned about not breaking other devices that might send their capabilities back differently.)

Reply:
We have to rely on the OpenRazer API - don't worry... I plan to test it against my Razer devices aswell